src/js/components/OptionalLabelsComponent.jsx

```diff
@@ -0,0 +1,196 @@
+var classNames = require("classnames");
+var React = require("react/addons");
+
+var Util = require("../helpers/Util");
+
+var AppFormErrorMessages = require("../constants/AppFormErrorMessages");
+var DuplicableRowControls = require("../components/DuplicableRowControls");
+var FormActions = require("../actions/FormActions");
+var FormGroupComponent =
+  require("../components/FormGroupComponent");
+
+var OptionalLabelsComponent = React.createClass({
+  displayName: "OptionalLabelsComponent",
+
+  propTypes: {
+    errorIndices: React.PropTypes.array,
+    generalError: React.PropTypes.string,
+    rows: React.PropTypes.array
+  },
+
+  populateInitialConsecutiveKeys: function (rows) {
```

Comment:
We could create a Mixin for this, that could read from `getDefaultProps()` the "shape" of each row and do the boilerplate stuff.